Sources/TuistCore/AsyncQueue/AsyncQueueDispatcher.swift

```diff
@@ -0,0 +1,16 @@
+import Foundation
+import RxSwift
+
+/// Async queue dispatcher.
+public protocol AsyncQueueDispatcher {
```

Comment:
@RomainBoulay assuming the `dispatchPersisted` will de-serialize the data into an instance of `AsyncQueueEvent` using the type associated with this dispatcher. Do you think we could define a protocol extension on `AsyncQueueDispatcher` to provide a default implementation? Or do you think we'd end up with the same issue that we ran into with the usage of generics and protocols that have an associated type?